data/blog: configuring Emacs for MDX files (#46)
tailscale-dev/tailscale-dev: data/blog/configuring-emacs-mdx.mdx

---
title: Configuring Emacs for MDX files
date: 2023-02-03

> authors: ['xe']

Reviewer: Almost forgot. Can you add a `summary` here?

Author: Done in https://github.com/tailscale-dev/tailscale-dev/pull/46/commits/77a7bda33158bb8b87ff6b8860e6e00eb0c4c2d2

tags:
  - 'emacs'
  - 'mdx'
---

I'm an [Emacs](https://www.gnu.org/software/emacs/) user and I have
been for the last decade. I use emacs for _everything_ from code, to
posts on this blog, even down to my daily TODO list with [Org
Mode](https://orgmode.org/). Naturally, I want to also use emacs to
edit posts on this blog. The only problem is that the blog uses
[MDX](https://mdxjs.com/) instead of normal Markdown. There isn't an
Emacs major mode for MDX and the [ticket for editor support in MDX was
closed](https://github.com/mdx-js/mdx/issues/119). This should mean
that I'm out of luck and need to architect a whole new major mode for

> Emacs.

Reviewer: ```suggestion
Emacs?
```

> closed](https://github.com/mdx-js/mdx/issues/119). This should mean
> that I'm out of luck and need to architect a whole new major mode for
> Emacs.

Reviewer: Feel free to ignore it. Just something I stumbled on when reading.

```suggestion
closed](https://github.com/mdx-js/mdx/issues/119). This should mean that I'm out of luck and must architect a new major mode for Emacs.
```
